Tests/Integration/MSFT_xFirewall.Integration.Tests.ps1

```diff
@@ -37,50 +70,82 @@ if ($executionPolicy -ne 'Unrestricted')
+    #>
+    $ConfigFile = Join-Path -Path $PSScriptRoot -ChildPath "$DSCResourceName.Config.ps1"
+    . $ConfigFile
+    
+    Describe "$($DSCResourceName)_Integration" {
+        #region DEFAULT TESTS
         It 'Should compile without throwing' {
             {
                 [System.Environment]::SetEnvironmentVariable('PSModulePath',
                     $env:PSModulePath,[System.EnvironmentVariableTarget]::Machine)
-                Firewall -OutputPath $env:Temp\Firewall
-                Start-DscConfiguration -Path $env:Temp\Firewall -ComputerName localhost -Wait -Verbose -Force
+                Invoke-Expression -Command "$($DSCResourceName)_Config -OutputPath `$WorkingFolder"
+                Start-DscConfiguration -Path (Join-Path -Path $env:Temp -ChildPath $DSCResourceName) `
+                    -ComputerName localhost -Wait -Verbose -Force
             } | Should not throw
         }
 
         It 'should be able to call Get-DscConfiguration without throwing' {
-            {Get-DscConfiguration} | Should Not throw
+            { Get-DscConfiguration -Verbose -ErrorAction Stop } | Should Not throw
         }
+        #endregion
 
         It 'Should have set the firewall and all the parameters should match' {
+            # Get the Rule details
             $firewallRule = Get-NetFireWallRule -Name $rule.Name
+            $Properties = @{
+                AddressFilters       = @(Get-NetFirewallAddressFilter -AssociatedNetFirewallRule $FirewallRule)
+                ApplicationFilters   = @(Get-NetFirewallApplicationFilter -AssociatedNetFirewallRule $FirewallRule)
+                InterfaceFilters     = @(Get-NetFirewallInterfaceFilter -AssociatedNetFirewallRule $FirewallRule)
+                InterfaceTypeFilters = @(Get-NetFirewallInterfaceTypeFilter -AssociatedNetFirewallRule $FirewallRule)
+                PortFilters          = @(Get-NetFirewallPortFilter -AssociatedNetFirewallRule $FirewallRule)
+                Profile              = @(Get-NetFirewallProfile -AssociatedNetFirewallRule $FirewallRule)
+                SecurityFilters      = @(Get-NetFirewallSecurityFilter -AssociatedNetFirewallRule $FirewallRule)
+                ServiceFilters       = @(Get-NetFirewallServiceFilter -AssociatedNetFirewallRule $FirewallRule)
+            }
 
-            $firewallRule.Name         | Should Be $rule.Name
-            $firewallRule.DisplayName  | Should Be $rule.DisplayName
-            $firewallRule.Group        | Should Be $rule.Group
-            $firewallRule.Enabled      | Should Be $rule.Enabled
-            $firewallRule.Profile      | Should Be $rule.Profile
-            $firewallRule.Action       | Should Be $rule.Action
-            $firewallRule.Description  | Should Be $rule.Description
-            $firewallRule.Direction    | Should Be $rule.Direction
+            # Use the Parameters List to perform these tests
+            foreach ($p in $ParameterList)
```

Comment:
Sorry about that! Done. As a matter of interest, what tools do you use for PS dev? I originally used VS 2013 wish POSH tools (and still do for some projects), but now I find myself using ISE (with ISE Steroids). I'd love it if VS Code had PS support though simply because how fast and light it is. :)